gnoi/os/server_test.go

```diff
@@ -151,28 +179,196 @@ func TestTargetActivateAndVerify(t *testing.T) {
+				result:   make(chan *pb.InstallResponse, 1),
+				errorReq: &pb.InstallRequest{Request: nil}, // Unexpected request.
+				os:       oS,
+			},
+			want: &installResult{
+				res: nil,
+				err: errors.New("Failed to receive TransferRequest"),
+			},
+		},
+		{
+			stream: &mockTransferStream{
+				response: make(chan *pb.InstallResponse, 1),
+				result:   make(chan *pb.InstallResponse, 1),
+				os: &mockos.OS{MockOS: mockosPb.MockOS{
+					Version: "1.0.0a",
+				}},
+			},
+			want: &installResult{
+				res: &pb.InstallResponse{Response: &pb.InstallResponse_InstallError{
+					InstallError: &pb.InstallError{Type: pb.InstallError_INSTALL_RUN_PACKAGE},
+				}},
+				err: errors.New("Attempting to force transfer an OS of the same version as the currently running OS"),
+			},
+		},
+		{
+			stream: &mockTransferStream{
+				response: make(chan *pb.InstallResponse, 1),
+				result:   make(chan *pb.InstallResponse, 1),
+				os: &mockos.OS{MockOS: mockosPb.MockOS{
+					Version: "1.0.3c",
+				}},
+			},
+			want: &installResult{
+				res: &pb.InstallResponse{Response: &pb.InstallResponse_Validated{Validated: &pb.Validated{Version: "1.0.3c"}}},
+				err: nil,
+			},
+		},
+		{
+			stream: &mockTransferStream{
+				response: make(chan *pb.InstallResponse, 1),
+				result:   make(chan *pb.InstallResponse, 1),
+				os: &mockos.OS{MockOS: mockosPb.MockOS{
+					Version: "1.0.2b",
+					Cookie:  "cookiestring",
+					Padding: buf,
+					Hash:    []byte("BADHASH"),
+				}},
+			},
+			want: &installResult{
+				res: &pb.InstallResponse{Response: &pb.InstallResponse_InstallError{InstallError: &pb.InstallError{Type: pb.InstallError_INTEGRITY_FAIL}}},
+				err: nil,
+			},
+		},
+		{
+			stream: &mockTransferStream{
+				response: make(chan *pb.InstallResponse, 1),
+				result:   make(chan *pb.InstallResponse, 1),
+				os:       incompatibleOS,
+			},
+			want: &installResult{
+				res: &pb.InstallResponse{Response: &pb.InstallResponse_InstallError{InstallError: &pb.InstallError{Type: pb.InstallError_INCOMPATIBLE, Detail: "Unsupported OS Version"}}},
+				err: nil,
+			},
+		},
+		{
+			stream: &mockTransferStream{
+				response: make(chan *pb.InstallResponse, 1),
+				result:   make(chan *pb.InstallResponse, 1),
+				os: &mockos.OS{MockOS: mockosPb.MockOS{
+					Version: "1.0.2c",
+				}},
+			},
+			want: &installResult{
+				res: &pb.InstallResponse{Response: &pb.InstallResponse_InstallError{InstallError: &pb.InstallError{Type: pb.InstallError_PARSE_FAIL}}},
+				err: nil,
+			},
+		},
+	}
+	for _, test := range tests {
+		server := NewServer(&Settings{FactoryVersion: "1.0.0a", InstalledVersions: []string{"1.0.3c"}})
+		got := &installResult{
+			err: server.Install(test.stream),
+		}
+		close(test.stream.result)
+		got.res = <-test.stream.result
+		if diff := pretty.Compare(test.want, got); diff != "" {
+			t.Errorf("Install(stream pb.OS_InstallServer): (-want +got):\n%s", diff)
+		}
+	}
+}
+
+func TestMultipleInstalls(t *testing.T) {
```

Comment:
because this test does not have multiple named cases, I would add a comment to clarify what tests and why it is important.

Reply:
Sure, that makes sense!

Reply:
Added